stl/inc/__msvc_int128.hpp

```diff
@@ -0,0 +1,1155 @@
+    }
+
+    _NODISCARD friend constexpr _Int128 operator+(const _Int128& _Left, const _Int128& _Right) noexcept {
+        return _Int128{static_cast<const _Uint128&>(_Left) + static_cast<const _Uint128&>(_Right)};
+    }
+    constexpr _Int128& operator+=(const _Int128& _That) noexcept {
+        _Uint128::operator+=(static_cast<const _Uint128&>(_That));
+        return *this;
+    }
+    template <integral _Ty>
+    friend constexpr _Ty& operator+=(_Ty& _Left, const _Int128& _Right) noexcept {
+        _Left = static_cast<_Ty>(_Int128{_Left} + _Right);
+        return _Left;
+    }
+
+    _NODISCARD friend constexpr _Int128 operator-(const _Int128& _Left, const _Int128& _Right) noexcept {
+        return _Int128{static_cast<const _Uint128&>(_Left) - static_cast<const _Uint128&>(_Right)};
+    }
+
+    constexpr _Int128& operator-=(const _Int128& _That) noexcept {
+        _Uint128::operator-=(static_cast<const _Uint128&>(_That));
+        return *this;
+    }
+    template <integral _Ty>
+    friend constexpr _Ty& operator-=(_Ty& _Left, const _Int128& _Right) noexcept {
+        _Left = static_cast<_Ty>(_Int128{_Left} - _Right);
+        return _Left;
+    }
+
+    _NODISCARD friend constexpr _Int128 operator*(_Int128 _Left, _Int128 _Right) noexcept {
+        bool _Negative = false;
+        if ((_Left._Word[1] & (1ull << 63)) != 0) {
+            _Left     = -_Left;
+            _Negative = true;
+        }
+        if ((_Right._Word[1] & (1ull << 63)) != 0) {
+            _Right    = -_Right;
+            _Negative = !_Negative;
+        }
+
+        _Uint128 _Result = static_cast<_Uint128&>(_Left) * static_cast<_Uint128&>(_Right);
+        if (_Negative) {
+            _Result = -_Result;
+        }
+        return _Int128{_Result};
+    }
+
+    constexpr _Int128& operator*=(const _Int128& _That) noexcept {
+        *this = *this * _That;
+        return *this;
+    }
+    template <integral _Ty>
+    friend constexpr _Ty& operator*=(_Ty& _Left, const _Int128& _Right) noexcept {
+        _Left = static_cast<_Ty>(_Int128{_Left} * _Right);
+        return _Left;
+    }
+
+    _NODISCARD friend constexpr _Int128 operator/(_Int128 _Num, _Int128 _Den) noexcept {
+        bool _Negative = false;
+        if ((_Num._Word[1] & (1ull << 63)) != 0) {
+            _Num      = -_Num;
+            _Negative = true;
+        }
+        if ((_Den._Word[1] & (1ull << 63)) != 0) {
+            _Den      = -_Den;
+            _Negative = !_Negative;
+        }
+
+        _Uint128 _Result = static_cast<_Uint128&>(_Num) / static_cast<_Uint128&>(_Den);
+        if (_Negative) {
+            _Result = -_Result;
+        }
+        return _Int128{_Result};
+    }
+    _NODISCARD friend constexpr _Int128 operator/(_Int128 _Num, const uint64_t _Den) noexcept {
+        bool _Negative = false;
+        if ((_Num._Word[1] & (1ull << 63)) != 0) {
+            _Num      = -_Num;
+            _Negative = true;
+        }

```

Comment:
Suggestion: This occurs several times (sometimes in pairs, sometimes alone). Should there be a helper function to handle "flip the value and the `_Negative` `bool`" which could be called repeatedly? Feel free to disregard.